vespa/deployment.py

```diff
@@ -440,65 +452,149 @@ def __init__(
-                serialization.PublicFormat.SubjectPublicKeyInfo,
+        self.control_plane_auth_method = None  # "api_key" or "access_token"
+        self.control_plane_access_token = None
+        self.auth_file_path = VESPA_HOME / "auth.json"
+        if self.api_key:
+            self.api_public_key_bytes = standard_b64encode(
+                self.api_key.public_key().public_bytes(
+                    serialization.Encoding.PEM,
+                    serialization.PublicFormat.SubjectPublicKeyInfo,
+                )
+            )
+            self.control_plane_auth_method = "api_key"
+        else:
+            self.api_public_key_bytes = None
+            self.control_plane_auth_method = "access_token"
+            print(
+                "No api-key found for control plane access. Using access token.",
+                file=self.output,
+            )
+            self.control_plane_access_token = self._try_get_access_token()
+            print(
+                "Successfully obtained access token for control plane access.",
+                file=self.output,
             )
-        )
         self.data_cert_path = None
         self.data_key_path = None
         self.data_key, self.data_certificate = self._load_certificate_pair()
         self.connection = http.client.HTTPSConnection(
             "api.vespa-external.aws.oath.cloud", 4443
         )
-        self.output = output_file
         self.auth_client_token_id = auth_client_token_id
         if auth_client_token_id is not None:
-            self.application_package.auth_clients = [
-                AuthClient(
-                    id="mtls",
-                    permissions=["read,write"],
-                    parameters=[
-                        Parameter("certificate", {"file": "security/clients.pem"})
-                    ],
-                ),
-                AuthClient(
-                    id="token",
-                    permissions=["read,write"],
-                    parameters=[Parameter("token", {"id": auth_client_token_id})],
-                ),
-            ]
+            if self.application_package is not None:
+                self.application_package.auth_clients = [
+                    AuthClient(
+                        id="mtls",
+                        permissions=["read,write"],
+                        parameters=[
+                            Parameter("certificate", {"file": "security/clients.pem"})
+                        ],
+                    ),
+                    AuthClient(
+                        id="token",
+                        permissions=["read,write"],
+                        parameters=[Parameter("token", {"id": auth_client_token_id})],
+                    ),
+                ]
+            else:
+                print(
+                    "Warning: Auth client token id set, but no application package provided. Make sure that services.xml is configured to use the provided token_id.",
```

Comment:
Is this really a warning, though? Is it not intended that users have a pre-existing package, and token auth? Certainly worth mentioning, but perhaps at a lower "level"?

Reply:
Good point. Reducing the level by removing "Warning" 👍